src/feasibility_checker.jl

```diff
@@ -0,0 +1,214 @@
+#  Copyright 2017, Iain Dunning, Joey Huchette, Miles Lubin, and contributors
+#  This Source Code Form is subject to the terms of the Mozilla Public
+#  License, v. 2.0. If a copy of the MPL was not distributed with this
+#  file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+using LinearAlgebra
+
+"""
+    primal_feasibility_report(
+        model::Model,
+        [point::AbstractDict{VariableRef,Float64}];
+        atol::Float64 = 0.0,
+        default::Union{Float64,Missing} = 0.0,
+    )::Dict{Any,Float64}
+
+Given a dictionary `point`, which maps variables to primal values, return a
+dictionary mapping the constraint reference of each constraint in `model` to the
+distance between the point and the nearest feasible point, if the distance is
+greater than `atol`.
+
+## Notes
+
+ * If a variable is not given in `point`, the value is assumed to be `default`.
+   If `default = missing`, constraints which contain a missing variable are
+   skipped.
+ * If no point is provided, the primal solution from the last time the model was
+   solved is used.
+
+## Examples
+
+```jldoctest; setup=:(using JuMP)
+julia> model = Model();
+
+julia> @variable(model, 0.5 <= x <= 1);
+
+julia> primal_feasibility_report(model, Dict(x => 0.2))
+Dict{Any,Float64} with 1 entry:
+  x ≥ 0.5 => 0.3
+```
+"""
+function primal_feasibility_report(
+    model::Model,
+    point::AbstractDict{VariableRef,Float64};
+    atol::Float64 = 0.0,
+    default::Union{Float64,Missing} = 0.0,
+)
+    point_f = x -> get(point, x, default)
+    violated_constraints = Dict{Any,Float64}()
+    for (F, S) in list_of_constraint_types(model)
+        _add_infeasible_constraints(
+            model, F, S, violated_constraints, point_f, atol
+        )
+    end
+    if num_nl_constraints(model) > 0
+        if ismissing(default)
+            error(
+                "`default` cannot be `missing` when nonlinear constraints " *
+                "are present.",
+            )
```

Comment:
why not?

Reply:
There are checks like this: https://github.com/jump-dev/JuMP.jl/blob/2792fab24726eca47660ceb332d9d27c89ebf80e/src/nlp.jl#L779
```Julia
julia> x = [missing]
1-element Array{Missing,1}:
 missing

julia> x == x
missing
```